src/scmdata/run.py

```diff
@@ -1379,16 +1392,24 @@ def interpolate(
             :obj:`target_times` grid
         """
         # pylint: disable=protected-access
+        target_times: TimePoints = TimePoints(target_times)
+        source_times: TimePoints = self.time_points
 
-        target_times = np.asarray(target_times, dtype="datetime64[s]")
+        if uniform_year_length:
+            source_time_values = source_times.years()
 
-        res = self.copy()
+            if len(np.unique(source_time_values)) != len(source_times):
+                raise ValueError("Non-unique year values with uniform_year_length=True")
+            target_time_values = target_times.years()
+        else:
+            source_time_values = source_times.values
+            target_time_values = target_times.values
 
-        target_times = TimePoints(target_times)
+        res = self.copy()
```

Comment:
Why is the copy needed?

Reply:
The operation is not inplace. The underlying data structures of `res` are updated below with the new data, but the metadata remains untouched.

Without the copy, it would be an in-place operation